src/movies.js

```diff
@@ -1,22 +1,72 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+function getAllDirectors(moviesArray) {
+  return moviesArray.map(function (element) {
+    return element.director;
+  });
+}
+
+const directors = getAllDirectors(movies);
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  let stephenMovies = moviesArray.filter(function (movie) {
+    return movie.director === "Steven Spielberg" && movie.genre === "Drama";
+  });
+
+  if (moviesArray.length === 0) {
+    return 0;
+  } else if (stephenMovies.length === 0) {
+    return 0;
+  } else {
+    return stephenMovies.length;
+  }
+}
+
+const dramaMovies = howManyMovies(movies);
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
+
 function scoresAverage(moviesArray) {}
```

Comment:
Here probably you can use a reduce function and inside of the reduce make sure that the property array from the currentElement exists in order tu add it to the total, if not just return the total